Services/UI/OJS.Services.Ui.Business/Implementations/ProblemsBusinessService.cs

```diff
@@ -137,8 +137,12 @@ public async Task<ProblemSearchServiceResultModel> GetSearchProblemsByName(Searc
         {
             var modelResult = new ProblemSearchServiceResultModel();
 
-            var allProblemsQueryable = this.problemsData.GetAllNonDeletedProblems()
-                .Where(p => p.Name.Contains(model.SearchTerm!));
+            var allProblemsQueryable = this.problemsData
+                .GetAllNonDeletedProblems()
+                .Where(p => p.Name.Contains(model.SearchTerm!) &&
+                            p.ProblemGroup.Contest.IsVisible &&
+                            (p.ProblemGroup.Contest.Category != null &&
+                             p.ProblemGroup.Contest.Category.IsVisible));
```

Comment:
Shouldn't be
```
(p.ProblemGroup.Contest.Category == null ||
 p.ProblemGroup.Contest.Category.IsVisible)
```

If we allow to have visible contests without category, then we should allow to have visible problems in such contests

Reply:
Currently you cannot create a Contest without setting a category at all:

<img width="677" alt="image" src="https://github.com/SoftUni-Internal/OpenJudgeSystem/assets/100528439/d51f7a6c-bf00-42ca-ae44-48aad7772916">

Null check was added since ContestCategory property is nullable in Contest model and its underlining Category.IsVisible check with "Possible null reference return" compiler warning.